File: Back-End/app/Models/Post.php
<?php

namespace App\Models;

use App\Models\Auth\Client;
use App\Models\Proposal;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Post extends Model
{
    use HasFactory;

    protected $fillable = [
        'client_id',
        'category_id',
        'team_id',
        'title',
        'description',
        'type',
        'budget',
        'deadline',
    ];

    public function client()
    {
        return $this->belongsTo(Client::class);
    }

    public function proposals()
    {
        return $this->hasMany(Proposal::class);
    }

    public function skills()
    {
        return $this->belongsToMany(Skill::class);
    }

    public function isProject()
    {
        return $this->type === 'project';
    }

    public function isJob()
    {
        return $this->type === 'job';
    }
}

File: Front-End/lib/models/post.dart
class Post {
  final int? id;
  final String title;
  final String description;
  final String deadline;
  final double budget;
  final String type;
  final int category;

  Post({
    this.id,
    required this.title,
    required this.description,
    required this.deadline,
    required this.budget,
    required this.type,
    required this.category
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      budget: json['budget'],
      deadline: json['deadline'],
      type: json['type'],
      category: json['category'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'budget': budget,
      'deadline': deadline,
      'type': type,
      'category':category,
    };
  }
}
